server.js

```diff
@@ -9,14 +11,57 @@ let fruits = [
   { id: "5", name: 'Orange', color: 'orange'}
 ]
 
+// @ GET - Display all
 app.get('/fruits', (req, res) => {
-  res.json(fruits)
+  const sortFruits = fruits.sort((a,b)=> a.id-b.id)
+  res.json(sortFruits)
 });
 
+// @ GET single fruit
 app.get('/fruits/:id', (req, res) => {
-  res.json({})
+  const {id} = req.params;
+  const findFruit = fruits.find(item=> item.id === id);
+  res.json(findFruit)
+})
+
+// @ POST - Add new fruit
+app.post("/fruits/add", (req,res)=>{
+  const{name} =req.body;
+  const checkName = fruits.find(fruit=>fruit.name === name)
+  if(checkName){
+    res.json({message: "Fruit already stored in Database!"})
+    return;
+  }
+  const newFruit = req.body;
+  fruits.push(newFruit);
+  res
+  .redirect("/fruits")
+  console.log("New fruit added succesfully!");
+})
+
+// @ PATCH - Update fruit
+
+app.patch("/fruits/:id", (req,res)=>{
+
+  const {id} = req.params;
+  const findFruit = fruits.find(fruit=>fruit.id == id);
+  if(!findFruit){
+    res.json({message: "Fruit not found"})
+    return;
+  }
+  const update = findFruit && Object.assign(findFruit, {...req.body});
```

Comment:
Nice hack :D 

Just a minor: The "findFruit" check in line 52 is actually not necesarry. Because it will always be defined. If it would be not defined, you would already have exited the route in line 50 :) 

But I like that you experiment with setups ;)